### ISSUE TITLE
When I tried to open a molecule with a PubChem CID greater than five characters molstar gave an error

### ISSUE THREAD

**Issue by IlyStra** [molstar/molstar#1354]
When I tried to open a molecule with a PubChem CID greater than five characters molstar gave an error.
**CID of molecule:** [5287971](https://pubchem.ncbi.nlm.nih.gov/compound/5287971)
**Log:**
![image](https://github.com/user-attachments/assets/804243fc-20ad-453f-a158-600285375c89)

**Comment by papillot** [comment]
I can confirm: a request is being made with the parameter `record_type=3d` and the pubchem API returns a `404 not found` response:

<img width="1143" alt="image" src="https://github.com/user-attachments/assets/c023322b-537a-410b-b386-cf83beb76339">

**Comment by dsehnal** [comment]
Yeah, this indeed seems the 3D record not being available as https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/CID/5287971/record/SDF/?record_type=2d works

**Comment by arose** [comment]
Should we support the 2d endpoint? Is there a way to decide what is available? Anyone want to implement it?

**Comment by papillot** [comment]
> Should we support the 2d endpoint? 
I don't think that we should:
- I can't see how 2D structures could be useful in 3D. They should be minimized and there is no easy way to do that, or to ensure that the result is good enough when dealing with large and or very flexible compounds (in the example for lecithin that is reported in this ticket, there are 40 rotatable bonds, mostly related with the aliphatic tails).
- when an SDF is transformed into a Mol* atomic 3D model, the values that specify stereo (atom parity and/or bonds up/down) in 2D world are lost, which makes re-exporting the data as an SDF, unstable (stereo is lost). At least this is what we had experienced with, in our implementation. This could be alleviated by pushing coordinates a tiny bit off the z===0 plane to reflect stereo, but still, this does not make the processing stable.

> Is there a way to decide what is available? 
PubChem has this REST API. There is a possibility to get a list of conformers for a record: https://pubchem.ncbi.nlm.nih.gov/docs/pug-rest#section=Conformers

**Comment by dsehnal** [comment]
We could always add a parameter to define the source endpoint (2D/3D; default 3D) to at least give a choice?
